src/Compilers/CSharp/Test/Semantic/Semantics/InterpolationTests.cs

```diff
@@ -11019,5 +11019,101 @@ public static class MemoryExtensions
             var comp = CreateCompilation(new[] { code, InterpolatedStringHandlerAttribute, InterpolatedStringHandlerArgumentAttribute }, targetFramework: TargetFramework.NetCoreApp);
             comp.VerifyDiagnostics();
         }
+
+        [Fact, WorkItem(54703, "https://github.com/dotnet/roslyn/issues/54703")]
+        public void BracesAreEscaped_01()
+        {
+            var code = @"
+int i = 1;
+System.Console.WriteLine($""{{ {i} }}"");";
+
+            var comp = CreateCompilation(new[] { code, GetInterpolatedStringHandlerDefinition(includeSpanOverloads: false, useDefaultParameters: false, useBoolReturns: false) });
+
+            var verifier = CompileAndVerify(comp, expectedOutput: @"
+{ 
+value:1
+ }");
+
+            verifier.VerifyIL("<top-level-statements-entry-point>", @"
+{
+  // Code size       56 (0x38)
+  .maxstack  3
+  .locals init (int V_0, //i
+                System.Runtime.CompilerServices.DefaultInterpolatedStringHandler V_1)
+  IL_0000:  ldc.i4.1
+  IL_0001:  stloc.0
+  IL_0002:  ldloca.s   V_1
+  IL_0004:  ldc.i4.6
```

Comment:
Should this still be 6? Given that the handler only ever sees the single brace shouldn't we be counting the `{{` and `}}` as 1 when calculating the length here?

Reply:
Thanks for catching this, Jared.
Fred, consider adding a more explicit validation for this (having the constructor log the value).